flux-refineck/src/checker.rs

```diff
@@ -794,7 +794,7 @@ impl<'a, 'tcx, P: Phase> Checker<'a, 'tcx, P> {
                     }
                 }
             }
-            CastKind::FloatToInt | CastKind::IntToFloat => {
+            CastKind::FloatToInt | CastKind::IntToFloat | CastKind::Pointer(_) => {
```

Comment:
Better to be explicit here to avoid problems in the future

```suggestion
            CastKind::FloatToInt | CastKind::IntToFloat | CastKind::Pointer(PointerCast::MutToConstPointer) => {
```